rtl/eclrtl/rtlfield.cpp

```diff
@@ -2617,14 +3104,60 @@ size32_t RtlDictionaryTypeInfo::deserialize(ARowBuilder & builder, IRowDeseriali
     }
 }
 
-
 void RtlDictionaryTypeInfo::readAhead(IRowDeserializerSource & in) const
 {
     size32_t size = in.readSize();
     in.skip(size);
 }
 
 
+int RtlDictionaryTypeInfo::compare(const byte * left, const byte * right) const
+{
+    if (isLinkCounted())
+    {
+        const size32_t countLeft = rtlReadSize32t(left);
+        const size32_t countRight = rtlReadSize32t(right);
+        const byte * * rowsLeft = (const byte * *)((const byte *)left + sizeof(size32_t));
+        const byte * * rowsRight = (const byte * *)((const byte *)right + sizeof(size32_t));
+        size32_t row = 0;
+        for (;;)
+        {
+            if ((row == countLeft) || (row == countRight))
+            {
+                if (countLeft == countRight)
+                    return 0;
+                if (row == countLeft)
+                    return -1;
+                else
+                    return +1;
+            }
+
+            //MORE: Is this correct
```

Comment:
I don't know but should probably decide before merging the code!

Reply:
It isn't - I will fix